File: 2433-find-the-original-array-of-prefix-xor/2433-find-the-original-array-of-prefix-xor.py
class Solution:
    def findArray(self, pref: List[int]) -> List[int]:
        result = [0 for i in range(len(pref))]
        result[0] = pref[0]
        now = pref[0]

        for i in range(1, len(pref)):
            result[i] = now ^ pref[i]
            now = pref[i]

        return result

File: 2433-find-the-original-array-of-prefix-xor/2433-find-the-original-array-of-prefix-xor.java
class Solution {
    public int[] findArray(int[] pref) {
        int[] result = new int[pref.length];
        int now = pref[0];
        result[0] = pref[0];

        for (int i = 1; i < pref.length; i++) {
            result[i] = now ^ pref[i];
            now = pref[i];
        }

        return result;
    }
}